com.unity.netcode.gameobjects/Runtime/NetworkVariable/CollectionSerializationUtility.cs

```diff
@@ -0,0 +1,664 @@
+            {
+                var val = value[i];
+                var prevVal = previousValue[i];
+                if (!NetworkVariableSerialization<T>.AreEqual(ref val, ref prevVal))
+                {
+                    ++numChanges;
+                    changes.Set(i);
+                }
+            }
+
+            for (var i = previousValue.Count; i < value.Count; ++i)
+            {
+                ++numChanges;
+                changes.Set(i);
+            }
+
+            if (numChanges >= value.Count * 0.9)
+            {
+                writer.WriteByteSafe(1);
+                NetworkVariableSerialization<List<T>>.Write(writer, ref value);
+                return;
+            }
+
+            writer.WriteByteSafe(0);
+            BytePacker.WriteValuePacked(writer, value.Count);
+            writer.WriteValueSafe(changes);
+            for (int i = 0; i < value.Count; ++i)
+            {
+                if (changes.IsSet(i))
+                {
+                    var reffable = value[i];
+                    if (i < previousValue.Count)
+                    {
+                        var prevReffable = previousValue[i];
+                        NetworkVariableSerialization<T>.WriteDelta(writer, ref reffable, ref prevReffable);
+                    }
+                    else
+                    {
+                        NetworkVariableSerialization<T>.Write(writer, ref reffable);
+                    }
+                }
+            }
+        }
+        public static void ReadListDelta<T>(FastBufferReader reader, ref List<T> value)
+        {
+            reader.ReadByteSafe(out byte full);
+            if (full == 1)
+            {
+                NetworkVariableSerialization<List<T>>.Read(reader, ref value);
+                return;
+            }
+            ByteUnpacker.ReadValuePacked(reader, out int length);
+            var changes = new ResizableBitVector(Allocator.Temp);
+            using var toDispose = changes;
+            {
+                reader.ReadNetworkSerializableInPlace(ref changes);
+
+                if (length < value.Count)
+                {
+                    value.RemoveRange(length, value.Count - length);
+                }
+
+                for (var i = 0; i < length; ++i)
+                {
+                    if (changes.IsSet(i))
+                    {
+                        if (i < value.Count)
+                        {
+                            T item = value[i];
+                            NetworkVariableSerialization<T>.ReadDelta(reader, ref item);
+                            value[i] = item;
+                        }
+                        else
+                        {
+                            T item = default;
+                            NetworkVariableSerialization<T>.Read(reader, ref item);
+                            value.Add(item);
+                        }
+                    }
+                }
+            }
+        }
+
```

Comment:
And just spot check from here down for areas that could use quick reference comments...